classes/template/PKPTemplateManager.inc.php

```diff
@@ -249,12 +250,78 @@ function setCacheability($cacheability = CACHEABILITY_PUBLIC) {
 	}
 
 	/**
+	 * Compile a LESS stylesheet
+	 *
+	 * @param $name string Unique name for this LESS stylesheet
+	 * @param $lessFile string Path to the LESS file to compile
+	 * @param $args array Optional arguments. SUpports:
+	 *   'baseUrl': Base URL to use when rewriting URLs in the LESS file.
+	 *   'addLess': Array of additional LESS files to parse before compiling
+	 * @return string Compiled CSS styles
+	 */
+	public function compileLess($name, $lessFile, $args = array()) {
+
+		// Load the LESS compiler
+		require_once('lib/pkp/lib/vendor/oyejorge/less.php/lessc.inc.php');
+		$less = new Less_Parser(array( 'relativeUrls' => false ));
+
+		$request = $this->_request;
+
+		// Allow plugins to intervene
+		HookRegistry::call('PageHandler::compileLess', array($request, $less, $name, $lessFile, $args));
```

Comment:
D'oh! Of course. :) I've thought about it a bit and gone with the following:

```
HookRegistry::call('PageHandler::compileLess', array(&$less, &$lessFile, &$args, $name, $request));
```

`$less` - This is the LESS compiler and they may want to run actions on it.
`$lessFile` - This is the output file and a likely candidate for modification.
`$args` - The most likely candidate for modifications, adding extra LESS files and such.
`$name` - This is a unique ID for the stylesheet and should never be modified
`$request` - I would think no one should be modifying this in a hook about compiling LESS files, but if they want to do so they can retrieve it directly.

Ok?